tools/validators/instance_validator/main.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import instance_parser 
+import sys
+# ONTOLOGY VALIDATION
+# import ontology_validation
+# import generate_universe
+
+# TODO add input and return type checks in all functions
+
+if __name__ == '__main__':
+    if len(sys.argv) != 2:
+        print('USAGE: python3 schema.py yaml_file_path')
+        sys.exit(1)
+    
+    # SYNTAX VALIDATION
+    print('\nValidator starting ...')
+    filename = sys.argv[1]
+
+    # throws errors for syntax
+    parsed = dict(instance_parser.parse_yaml(filename))
+
+    print('Passed syntax checks!')
+
+    # ONTOLOGY VALIDATION
+    '''
+    print('Building ontology universe ...')
+
+    universe = generate_universe.build_universe()
+    fields, subfields_map, states_map, units_map, entities_map = generate_universe.parse_universe(universe)
```

Comment:
generate universe does not appear to be imported.

Reply:
Yes, for this PR it's not imported because the ontology-related files are in a separate PR. Once all these initial PRs are imported, I'm going to make another PR with new testcases and combine all the elements into the working validator.